client/my-sites/site-settings/jetpack-module-toggle.jsx

```diff
@@ -31,7 +31,7 @@ class JetpackModuleToggle extends Component {
 	};
 
 	static propTypes = {
-		siteId: PropTypes.number.isRequired,
```

Comment:
Well, the rationale was that it was required when we thought we would never be handling a `null` value here. It was there before the usage of the redux site selectors was widespread among the settings tabs and a lot more render blocking techniques were used before reaching the JetpackModuleToggle rendering.

Looks to me that if we're making the whole system be able to handle a `null` value for sites or site ids, we should make this toggle component at least be able to render without warnings when that particular state is reached by the system. Also this non required usage of a siteId prop looks to be the case for 80% of the components that use a `siteId` as prop.

Otherwise, if we follow techniques like `{ siteId && <JetpackModuleToggle siteId={siteId} /> }` we would probably need to do that everywhere for consistency even when we don't detect a warning right now. For example, here in this case, it will demand that 3 parent components make that check. And that's just one of the tabs. 

What is stopping us from currently making every child component on every piece of Calypso be conditionally rendered if the props we're passing to it are null or such? This last rationale is what I was trying to avoid with this change.